app/store/fetchGamesReducer.js

```diff
@@ -0,0 +1,45 @@
+import apiService from '../services/ApiService'
+
+const initialState = {
+  games: [],
+  hasErrored: false,
+  isLoading: false
+};
+
+export default function fetchGamesReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'FETCH_GAMES':
+      return {
+        ...state,
+        isLoading: true
+      };
+    case 'FETCH_GAMES_COMPLETE':
+      return {
+        ...state,
+        isLoading: false,
+        games: action.payload
+      };
+    case 'FETCH_PLAYERS_ERROR':
+      return {
+        ...state,
+        isLoading: false,
+        hasErrored: true
+      };
+    default:
+      return state;
+  }
+}
+
+export const fetchGames = () => dispatch => {
+  dispatch({
+    type: 'FETCH_GAMES'
+  });
+
+  return apiService.fetchGames()
```

Comment:
In my Todos project I don't fetch data in my reducers - I only use them for manipulating state. I've a seperate file for 'action dispatchers' [(`/actions/`](https://github.com/janakerman/todos/blob/master/src/actions/index.js). I'm happy with this for now but would be good if you could you explain to me the best practice about where this should live / show me an article / ask Nick?